learning/tour-of-beam/terraform/build.gradle.kts

```diff
@@ -0,0 +1,388 @@
+        var environment = project.property("project_environment") as String
+        var gcloud_account = project.property("gcloud_account") as String
+        args(
+                "apply",
+                "-auto-approve",
+                "-lock=false",
+                "-parallelism=3",
+                "-var=pg_router_host=$pg_router_host",
+                "-var=gcloud_init_account=$gcloud_account",
+                "-var=environment=$environment",
+                if (file("./environment/$environment/terraform.tfvars").exists()) {
+                    "-var-file=./environment/$environment/terraform.tfvars"
+                } else {
+                    "-no-color"
+                }
+        )
+    tasks.getByName("uploadLearningMaterials").mustRunAfter(this)
+
+    }
+
+tasks.register<TerraformTask>("terraformDestroy") {
+    var pg_router_host = project.extensions.extraProperties["pg_router_host"] as String
+    var environment = project.property("project_environment") as String
+    var gcloud_account = project.property("gcloud_account") as String
+    args(
+            "destroy",
+            "-auto-approve",
+            "-lock=false",
+            "-var=pg_router_host=$pg_router_host",
+            "-var=environment=$environment",
+            "-var=gcloud_init_account=$gcloud_account",
+            if (file("./environment/$environment/terraform.tfvars").exists()) {
+                "-var-file=./environment/$environment/terraform.tfvars"
+            } else {
+                "-no-color"
+            }
+    )
+}
+
+tasks.register("getRouterHost") {
+    group = "backend-deploy"
+    val result = ByteArrayOutputStream()
+    exec {
+        commandLine("kubectl", "get", "svc", "-l", "app=backend-router-grpc", "-o", "jsonpath='{.items[0].status.loadBalancer.ingress[0].ip}:{.items[0].spec.ports[0].port}'")
+        standardOutput = result
+    }
+    val pg_router_host = result.toString().trim().replace("'", "")
+    project.extensions.extraProperties["pg_router_host"] = pg_router_host
+}
+
+tasks.register("indexcreate") {
+    group = "backend-deploy"
+    val indexpath = "../backend/internal/storage/index.yaml"
+    exec {
+        executable("gcloud")
+        args("datastore", "indexes", "create", indexpath)
+    }
+}
+
+tasks.register("firebaseProjectCreate") {
+    group = "frontend-deploy"
+    val result = ByteArrayOutputStream()
+    var project_id = project.property("project_id") as String
+    exec {
+        executable("firebase")
+        args("projects:list")
+        standardOutput = result
+    }
+    val output = result.toString().trim()
+    if (output.contains(project_id)) {
+        println("Firebase is already added to project $project_id.")
+    } else {
+        exec {
+            executable("firebase")
+            args("projects:addfirebase", project_id)
+        }.assertNormalExitValue()
+        println("Firebase has been added to project $project_id.")
+    }
+}
```

Comment:
Why can't someone just use the firebase cli directly?

Reply:
FIrebase CLI directly will be used now, as a sequence of commands, including Terraform, Firebase and Flutter
Kotlin Gradle script will no longer be involved into infrastructure creation and configuration